settlement-pipelines/src/bin/claim_settlement.rs

```diff
@@ -358,59 +339,54 @@ async fn main() -> anyhow::Result<()> {
                 }
             };
 
-            if stake_accounts_per_staker_cache
-                .get(&tree_node.stake_authority)
-                .is_none()
+            let stake_withdraw_pair = StakeWithdrawAuthorityPair::new(
```

Comment:
it would be much niceer if we had a `struct StakeAccountsCache` and you would just call `.get(witdraw_authority, stake_authority)`
then you would call `pick_stake_account_for_claim` which would give you account or tell you that only locked accounts are in there

Reply:
In full agreement, thanks a lot for pointing that out.